pkg/storage/s3/s3.go

```diff
@@ -238,36 +274,75 @@ func (d *driver) ConfigEnv() (envs envvar.List, err error) {
 }
 
 func (d *driver) Volumes() ([]corev1.Volume, []corev1.VolumeMount, error) {
-	if d.Config.CloudFront == nil {
-		return nil, nil, nil
-	}
+	volumes := []corev1.Volume{}
+	volumeMounts := []corev1.VolumeMount{}
 
 	optional := false
 
-	vol := corev1.Volume{
-		Name: "registry-cloudfront",
+	// Mount the registry config secret containing the credentials file data
+	credsVolume := corev1.Volume{
+		Name: defaults.ImageRegistryPrivateConfiguration,
 		VolumeSource: corev1.VolumeSource{
 			Secret: &corev1.SecretVolumeSource{
-				SecretName: d.Config.CloudFront.PrivateKey.Name,
-				Items: []corev1.KeyToPath{
-					{Key: d.Config.CloudFront.PrivateKey.Key, Path: "private.pem"},
-				},
-				Optional: &optional,
+				SecretName: defaults.ImageRegistryPrivateConfiguration,
+				Optional:   &optional,
 			},
 		},
 	}
 
-	mount := corev1.VolumeMount{
-		Name:      vol.Name,
-		MountPath: "/etc/docker/cloudfront",
+	credsVolumeMount := corev1.VolumeMount{
+		Name:      credsVolume.Name,
+		MountPath: imageRegistrySecretMountpoint,
 		ReadOnly:  true,
 	}
 
-	return []corev1.Volume{vol}, []corev1.VolumeMount{mount}, nil
+	volumes = append(volumes, credsVolume)
+	volumeMounts = append(volumeMounts, credsVolumeMount)
+
+	if d.Config.CloudFront != nil {
+		vol := corev1.Volume{
+			Name: "registry-cloudfront",
+			VolumeSource: corev1.VolumeSource{
+				Secret: &corev1.SecretVolumeSource{
+					SecretName: d.Config.CloudFront.PrivateKey.Name,
+					Items: []corev1.KeyToPath{
+						{Key: d.Config.CloudFront.PrivateKey.Key, Path: "private.pem"},
+					},
+					Optional: &optional,
+				},
+			},
+		}
+
+		volumes = append(volumes, vol)
+
+		mount := corev1.VolumeMount{
+			Name:      vol.Name,
+			MountPath: "/etc/docker/cloudfront",
+			ReadOnly:  true,
+		}
+
+		volumeMounts = append(volumeMounts, mount)
+	}
+
+	return volumes, volumeMounts, nil
 }
 
 func (d *driver) VolumeSecrets() (map[string]string, error) {
-	return nil, nil
+	// Return the same credentials data that the image-registry-operator is using
+	// so that it can be stored in the image-registry Pod's Secret.
+	cfg, err := d.GetConfig()
+	if err != nil {
+		return nil, fmt.Errorf("failed to retrieve S3 driver config: %v", err)
+	}
+	defer os.Remove(cfg.SharedCredentialsFile)
+
+	data, err := ioutil.ReadFile(cfg.SharedCredentialsFile)
```

Comment:
This is a strange exercise: create a file, read its content, and delete the file.
I don't have suggestions, but there is room for improvement.